### PR TITLE
[Typing][A-20,A-61] Add type annotations for `paddle/nn/{layer,functional}/loss.py`

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR Category

User Experience

### PR Types

Improvements

### Description

- #65008

### CODE REVIEW

**Review by SigureMo — COMMENTED** [review]
比较建议 functional 和相关 layer 一并推进，或者先完成 functional 再完成 layer，以确保两者代码风格的一致性，并且推荐在 functional 定义 Alias，并在 layer 中 import 复用，以保持合理的依赖关系

@Asthestarsfalll 方便的话这个 PR 里推进 loss 的 functional 和 layer 吧

cc @gsq7474741

**Comment by SigureMo** [comment]
#65379 这种大 PR 也是建议拆分成 functional-layer pair 的形式的，PR 越大越不容易推进合入，合理地合并 PR 更利于 review

**Comment by Asthestarsfalll** [comment]
> #65379 这种大 PR 也是建议拆分成 functional-layer pair 的形式的，PR 越大越不容易推进合入，合理地合并 PR 更利于 review

好的

**Review by SigureMo — DISMISSED** [review]
LGTMeow

**Comment by SigureMo** [comment]
```
2024-06-25 21:27:02 **************************************************************
2024-06-25 21:27:02 Please find RD for approval first, and then find TPM for approval.
2024-06-25 21:27:02 0. You must have one RD (XiaoguangHu01, jeff41404 or qingqing01) approval for API change.
2024-06-25 21:27:02 
2024-06-25 21:27:02 There are 1 approved errors.
2024-06-25 21:27:02 **************************************************************
```

怎么会有 API diff 呢？

**Comment by Asthestarsfalll** [comment]
> ```
> 2024-06-25 21:27:02 **************************************************************
> 2024-06-25 21:27:02 Please find RD for approval first, and then find TPM for approval.
> 2024-06-25 21:27:02 0. You must have one RD (XiaoguangHu01, jeff41404 or qingqing01) approval for API change.
> 2024-06-25 21:27:02 
> 2024-06-25 21:27:02 There are 1 approved errors.
> 2024-06-25 21:27:02 **************************************************************
> ```
> 
> 怎么会有 API diff 呢？

是因为参数默认值改了吗？我使用代码生成0.00001会变成1e-5

**Comment by SigureMo** [comment]
> 是因为参数默认值改了吗？我使用代码生成0.00001会变成1e-5

哪个 API 呢？

**Comment by SigureMo** [comment]
利用最新 commit 打印一下 diff 再看一下吧，现在看不出来

**Comment by SigureMo** [comment]
```
2024-06-26 00:03:13 api_spec_diff: API Difference is: 
2024-06-26 00:03:13 - paddle.nn.functional.triplet_margin_loss (ArgSpec(args=['input', 'positive', 'negative', 'margin', 'p', 'epsilon', 'swap', 'reduction', 'name'], varargs=None, varkw=None, defaults=(1.0, 2, 1e-06, False, 'mean', None), kwonlyargs=[], kwonlydefaults=None), ('document', '**********'))
2024-06-26 00:03:13 + paddle.nn.functional.triplet_margin_loss (ArgSpec(args=['input', 'positive', 'negative', 'margin', 'p', 'epsilon', 'swap', 'reduction', 'name'], varargs=None, varkw=None, defaults=(1.0, 2.0, 1e-06, False, 'mean', None), kwonlyargs=[], kwonlydefaults=None), ('document', '**********'))
2024-06-26 00:03:13 ?                                                                                                                                                                                            ++
```

找到了

**Review by SigureMo — APPROVED** [review]

**Review by SigureMo — DISMISSED** [review]

**Review by SigureMo — APPROVED** [review]
